sharedUiComponents/lines/checkBoxLineComponent.tsx

```diff
@@ -85,9 +88,11 @@ export class CheckBoxLineComponent extends React.Component<ICheckBoxLineComponen
         return (
             <div className="checkBoxLine">
                 {this.props.icon && <img src={this.props.icon} title={this.props.iconLabel} alt={this.props.iconLabel} className="icon" />}
-                <div className="label" title={this.props.iconLabel}>
-                    {this.props.label}
-                </div>
+                {this.props.label && 
+                    <div className="label" title={this.props.iconLabel}>
+                        {this.props.label}
+                    </div>}
+                {this.props.faIcons && <FontAwesomeIcon icon={this.state.isSelected ? this.props.faIcons.faIconEnabled : this.props.faIcons.faIconDisabled} onClick={() => !this.props.disabled && this.onChange()}/>}
                 <div className="checkBox">

```

Comment:
maybe instead of hiding it in css either show the icons of the actual checkbox. Would also be great to allow the disabled state even when icons are used.

Reply:
I put a conditional to not render the checkbox if the faIcons property isn't undefined, but I've left the checkBox classname "hidden" since the other components that use the checkbox use that class.